resolwe_bio/processes/support_processors/merge_samples.py

```diff
@@ -0,0 +1,206 @@
+"""Merge FASTQs."""
+import glob
+import os
+from pathlib import Path
+from shutil import move
+
+from plumbum import TEE
+from resolwe.process import (
+    Cmd,
+    DataField,
+    FileField,
+    FileHtmlField,
+    ListField,
+    Process,
+    SchedulingClass,
+)
+
+
+def run_fastqc(fastqs, output_dir):
+    """Run fastQC on given FASTQs.
+
+    :param list fastqs: List of fastqs
+    :param str output_dir: Output directory
+
+    """
+    output_path = Path(output_dir)
+    if output_path.exists() and not output_path.is_dir():
+        raise TypeError(f"'{output_dir}' already exists but is not a directory.")
+    elif not output_path.exists():
+        os.mkdir(str(output_path))
+
+    cmd = Cmd['fastqc']
+    for fastq in fastqs:
+        cmd = cmd[fastq]
+    cmd = cmd['--extract']
+    cmd = cmd[f'--outdir={str(output_path)}']
+    _, _, stderr = cmd & TEE
+
+    return stderr
+
+
+class MergeFastqSingle(Process):
+    """Merge single-end FASTQs into one sample."""
+    slug = 'merge-fastq-single'
+    name = 'Merge FASTQ (single-end)'
+    process_type = 'data:reads:fastq:single'
+    version = '1.0.0'
+    category = 'Other'
+    scheduling_class = SchedulingClass.BATCH
+    entity = {
+        'type': 'sample',
+        'descriptor_schema': 'sample',
+        'always_create': True,
+    }
+    requirements = {
+        'expression-engine': 'jinja',
+        'executor': {
+            'docker': {
+                'image': 'resolwebio/common:1.3.1'
+            }
+        },
+    }
+    data_name = '{{ reads|map("sample_name")|join(", ")|default("?") }}'
+
+    class Input:
+        """Input fields to process MergeFastqSingle."""
+        reads = ListField(
+            DataField(data_type='reads:fastq:single:'),
+            label='Reads data objects',
+        )
+
+    class Output:
+        """Output fields to process MergeFastqSingle."""
+        fastq = ListField(FileField(), label='Reads file')
+        fastqc_url = ListField(
+            FileHtmlField(),
+            label='Quality control with FastQC',
+        )
+        fastqc_archive = ListField(FileField(), label='Download FastQC archive')
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        fastqc = []
+        name = f'{Path(inputs.reads[0].fastq[0].path).stem}_merged'
+        merged_fastq = f'{name}.fastq.gz'
+        with open(merged_fastq, 'wb') as outfile:
+            for reads in inputs.reads:
+                for fastq in reads.fastq:
+                    with open(fastq.path, 'rb') as infile:
+                        for line in infile:
+                            outfile.write(line)
+
+        outputs.fastq = [merged_fastq]
+
+        print('Postprocessing FastQC...')
```

Comment:
This will print to the process state (info, warning, error) which is not the goal here. This print statement is there to have a better overview over the stdout output.